src/conventions/h-group/clue-interpretation/connecting-cards.js

```diff
@@ -107,22 +108,26 @@ export function find_known_connecting(game, giver, identity, ignoreOrders = [])
  * @param {number} target 			The player index receiving the clue. They will not find self-prompts or self-finesses.
  * @param {number} reacting
  * @param {Identity} identity
+ * @param {boolean} looksDirect 	Whether the clue could be interpreted as direct play (i.e. never as self-prompt/finesse).
  * @param {boolean} firstPlay		Is this the first play?
  * @param {number[]} [connected] 	The orders of cards that have previously connected (and should be skipped).
  * @param {number[]} [ignoreOrders] The orders of cards to ignore when searching.
  * @returns {Connection | undefined}
  */
-function find_unknown_connecting(game, giver, target, reacting, identity, firstPlay, connected = [], ignoreOrders = []) {
+function find_unknown_connecting(game, giver, target, reacting, identity, looksDirect, firstPlay, connected = [], ignoreOrders = []) {
 	const { common, state } = game;
 
 	const hand = state.hands[reacting];
 	const prompt = common.find_prompt(hand, identity, state.variant, connected, ignoreOrders);
 	const finesse = common.find_finesse(hand, connected, ignoreOrders);
 
+	// Track if the identity could also be in our own hand.
+	const self_connection = find_own_prompt_or_finesse(game.minimalCopy(), giver, target, identity, looksDirect, connected, ignoreOrders);
```

Comment:
These copies of the state could be removed if we make sure that find_own_prompt_or_finesse at least restores inferred states.

Reply:
I think the only mutation is in `resolved_layered_finesse()`. I think that code should probably be moved to `assign_connections()`, but it doesn't have to be in this PR.